frontend/device/generic/device.lua

```diff
@@ -500,57 +502,360 @@ function Device:exit()
+                        local ifr = ffi.new("struct ifreq")
+                        ffi.copy(ifr.ifr_ifrn.ifrn_name, ifa.ifa_name, C.IFNAMSIZ)
+                        if C.ioctl(socket, C.SIOCGIFHWADDR, ifr) == -1 then
+                            local errno = ffi.errno()
+                            logger.err("Device:retrieveNetworkInfo: SIOCGIFHWADDR ioctl:", ffi.string(C.strerror(errno)))
+                        else
+                            local mac = string.format("%02X:%02X:%02X:%02X:%02X:%02X",
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[0], 0xFF),
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[1], 0xFF),
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[2], 0xFF),
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[3], 0xFF),
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[4], 0xFF),
+                                                      bit.band(ifr.ifr_ifru.ifru_hwaddr.sa_data[5], 0xFF))
+                            table.insert(results, T(_("MAC: %1"), mac))
+                        end
+
+                        -- Check if it's a wireless interface (c.f., wireless-tools)
+                        local iwr = ffi.new("struct iwreq")
+                        ffi.copy(iwr.ifr_ifrn.ifrn_name, ifa.ifa_name, C.IFNAMSIZ)
+                        if C.ioctl(socket, C.SIOCGIWNAME, iwr) ~= -1 then
+                            interfaces[ifname] = "wireless"
+                            -- Get its ESSID
+                            local essid = ffi.new("char[?]", C.IW_ESSID_MAX_SIZE + 1)
+                            iwr.u.essid.pointer = ffi.cast("caddr_t", essid)
+                            iwr.u.essid.length = C.IW_ESSID_MAX_SIZE + 1
+                            iwr.u.essid.flags = 0
+                            if C.ioctl(socket, C.SIOCGIWESSID, iwr) == -1 then
+                                local errno = ffi.errno()
+                                logger.err("Device:retrieveNetworkInfo: SIOCGIWESSID ioctl:", ffi.string(C.strerror(errno)))
+                            else
+                                local essid_on = iwr.u.data.flags
+                                if essid_on ~= 0 then
+                                    local token_index = bit.band(essid_on, C.IW_ENCODE_INDEX)
+                                    if token_index > 1 then
+                                        table.insert(results, T(_("SSID: \"%1\" [%2]"), ffi.string(essid), token_index))
+                                    else
+                                        table.insert(results, T(_("SSID: \"%1\""), ffi.string(essid)))
+                                    end
+                                else
+                                    table.insert(results, _("SSID: off/any"))
```

Comment:
What exactly does `any` mean here?

Reply:
That's a fair question, I'm assuming that means a hidden SSID (that's the same string as iwconfig, FWIW).

Reply:
Ah, I didn't realize it was just copying the logic more or less verbatim. (first search result with the source, probably a bit outdated)

https://github.com/HewlettPackard/wireless-tools/blob/c1074342112d8a4cdc44275b1bc15701aaf7f30b/wireless_tools/iwconfig.c#L159-L185